### PR TITLE
Add information when throws UnavailableDataSourceException

### LINKED ISSUE #24152: Add source database information when throws UnavailableDataSourceException
## Feature Request

Add source database information when throws UnavailableDataSourceException.
抛出 UnavailableDataSourceException 时，附加异常的源数据库信息。

```
Exception in thread "main" org.apache.shardingsphere.infra.datasource.state.exception.UnavailableDataSourceException: DATA-SOURCE-00001: Data source unavailable.
    at org.apache.shardingsphere.infra.util.exception.ShardingSpherePreconditions.checkState(ShardingSpherePreconditions.java:41)
    at org.apache.shardingsphere.infra.datasource.state.DataSourceStateManager.checkState(DataSourceStateManager.java:87)
    at org.apache.shardingsphere.infra.datasource.state.DataSourceStateManager.initState(DataSourceStateManager.java:79)
    at org.apache.shardingsphere.infra.datasource.state.DataSourceStateManager.lambda$initStates$0(DataSourceStateManager.java:70)
    at java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
    at org.apache.shardingsphere.infra.datasource.state.DataSourceStateManager.initStates(DataSourceStateManager.java:70)
    at org.apache.shardingsphere.mode.metadata.MetaDataContextsFactory.lambda$checkDataSourceStates$4(MetaDataContextsFactory.java:114)
    at java.util.HashMap.forEach(HashMap.java:1290)
    at org.apache.shardingsphere.mode.metadata.MetaDataContextsFactory.checkDataSourceStates(MetaDataContextsFactory.java:112)
    at org.apache.shardingsphere.mode.metadata.MetaDataContextsFactory.create(MetaDataContextsFactory.java:88)
    at org.apache.shardingsphere.mode.manager.cluster.ClusterContextManagerBuilder.build(ClusterContextManagerBuilder.java:59)
    at org.apache.shardingsphere.proxy.initializer.BootstrapInitializer.createContextManager(BootstrapInitializer.java:68)
    at org.apache.shardingsphere.proxy.initializer.BootstrapInitializer.init(BootstrapInitializer.java:58)
    at org.apache.shardingsphere.proxy.Bootstrap.main(Bootstrap.java:52)
```

### PR DESCRIPTION
**Pull request by sunkai-cai**
Fixes #24152.

### CODE REVIEW

File: infra/common/src/main/java/org/apache/shardingsphere/infra/datasource/state/exception/UnavailableDataSourceException.java [hunk 1]
```diff
@@ -35,4 +35,8 @@ public final class UnavailableDataSourceException extends ShardingSphereServerEx
     public UnavailableDataSourceException(final SQLException cause) {
         super(ERROR_CATEGORY, ERROR_CODE, "Data source unavailable.", cause);
     }
+    
+    public UnavailableDataSourceException(final SQLException cause, final String databaseName) {
+        super(ERROR_CATEGORY, ERROR_CODE, String.format("Data source unavailable from `%s` .", databaseName), cause);
```
Comment (RaigorJiang) [inline infra/common/src/main/java/org/apache/shardingsphere/infra/datasource/state/exception/UnavailableDataSourceException.java:40]: Hi @sunkai-cai ,
Is there an extra space after `%s`?

Comment (sunkai-cai) [reply infra/common/src/main/java/org/apache/shardingsphere/infra/datasource/state/exception/UnavailableDataSourceException.java:40]: oh, i remove it.

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
Thank you! @sunkai-cai

**Comment by sunkai-cai** [comment]
@RaigorJiang  You are welcome :)

**Pull request merged**